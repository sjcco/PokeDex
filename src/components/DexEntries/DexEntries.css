
.container {
  height: 100%;
  display: flex;
  flex-direction: column;
  border-radius: 0.5rem;
  box-shadow: 0.05rem 0.1rem 0.3rem -0.03rem rgba(0, 0, 0, 0.45);
  margin: 0.5rem 0.5rem;
  background-color: white;
}

.container p {
  border: 1px solid rgb(197, 197, 197);
  border-radius: 0.25rem;
  width: 80%;
  margin: 1rem auto 0 auto;
  padding: 0.5rem 1.5rem;
  font-family: 'Trebuchet MS', Arial, sans-serif;
  font-weight: 200;
  font-size: 0.9rem;
  color: rgb(80, 79, 79);
  text-align: center;
}

.entry-description {
  margin: 0.5rem auto 1rem auto;
  width: auto;
  font-family: 'Trebuchet MS', Arial, sans-serif;
  font-weight: 100;
  font-size: 0.8rem;
  color: rgb(180, 180, 180);
}

.weight-height {
  display: flex;
  width: 90%;
  margin: 0 auto 1rem auto;
  justify-content: space-between;
}

.weight-container {
  width: 45%;
}

.height-container {
  width: 45%;
}

.data-display {
  width: auto;
  border: 1px solid rgb(197, 197, 197);
  border-radius: 0.25rem;
  margin: 1rem auto 0 auto;
  padding: 0.5rem 1.5rem;
  font-family: 'Trebuchet MS', Arial, sans-serif;
  font-weight: 200;
  font-size: 0.8rem;
  color: rgb(80, 79, 79);
  text-align: center;
}

.data-names {
  width: auto;
  margin: 0 auto;
  font-family: 'Trebuchet MS', Arial, sans-serif;
  font-weight: 100;
  font-size: 0.8rem;
  color: rgb(180, 180, 180);
  text-align: center;
}
