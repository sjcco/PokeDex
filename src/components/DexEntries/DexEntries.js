import React from 'react';
import PropTypes from 'prop-types';
import {
  container, weightHeight, weightContainer, heightContainer,
  dataDisplay, dataNames, entryDescription,
} from './DexEntries.css';

const DexEntries = ({ description, weight, height }) => (
  <div className={container}>
    <p>{description}</p>
    <h4 className={entryDescription}>PokeDex entry</h4>
    <div className={weightHeight}>
      <div className={weightContainer}>
        <h3 className={dataDisplay}>{weight}</h3>
        <h4 className={dataNames}>weight</h4>
      </div>
      <div className={heightContainer}>
        <h3 className={dataDisplay}>{height}</h3>
        <h4 className={dataNames}>height</h4>
      </div>
    </div>
  </div>
);

DexEntries.propTypes = {
  description: PropTypes.string.isRequired,
  weight: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired,
};

export default DexEntries;
