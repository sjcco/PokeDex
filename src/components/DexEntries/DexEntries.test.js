/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import DexEntries from './DexEntries';

test('renders content', () => {
  const component = renderer.create(<DexEntries
    description="it"
    weight="works"
    height="!!!"
  />).toJSON();
  expect(component).toMatchSnapshot();
});
