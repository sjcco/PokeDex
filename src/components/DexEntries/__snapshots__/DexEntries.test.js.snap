// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content 1`] = `
<div
  className="container"
>
  <p>
    it
  </p>
  <h4
    className="entryDescription"
  >
    PokeDex entry
  </h4>
  <div
    className="weightHeight"
  >
    <div
      className="weightContainer"
    >
      <h3
        className="dataDisplay"
      >
        works
      </h3>
      <h4
        className="dataNames"
      >
        weight
      </h4>
    </div>
    <div
      className="heightContainer"
    >
      <h3
        className="dataDisplay"
      >
        !!!
      </h3>
      <h4
        className="dataNames"
      >
        height
      </h4>
    </div>
  </div>
</div>
`;
